synapse/storage/schema/main/delta/59/14remove_hidden_devices_from_device_inbox.sql

```diff
@@ -0,0 +1,20 @@
+/* Copyright 2021 The Matrix.org Foundation C.I.C
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+--- Remove messages to hidden devices from the device_inbox table.
+--- This schould run as the last task, it may take a little bit longer
+--- to finish.
+
+DELETE FROM device_inbox WHERE device_id IN (SELECT device_id FROM devices WHERE hidden = TRUE);
```

Comment:
Heads up that old versions of sqlite will treat `TRUE` as the string "TRUE". We've recently written some documentation which explains this [here](https://matrix-org.github.io/synapse/develop/development/database_schema.html#boolean-columns).

But unfortunately we still support old affected SQLite versions. I realise we're removing this file for now - but once we introduce it back in we'll need to use statement arguments instead.